tests/unit/core/api/test_index.js

```diff
@@ -264,6 +302,38 @@ describe(__filename, () => {
           return mockWindow.verify();
         });
     });
+
+    it('does not change clientApp for Android extensions', () => {
```

Comment:
Once you move the workaround code into a shared function, I don't think you need to test it here because that would be a duplication of the tests in search().